cli/src/main/java/org/jboss/as/cli/impl/ReadlineConsole.java

```diff
@@ -638,49 +666,45 @@ private void logPromptMask(String prompt, Character mask) {
         LOG.tracef("Prompt %s mask %s", prompt, mask);
     }
 
-    private void interrupted(InterruptedException ex) {
-        LOG.trace("Thread interrupted");
-
-        // Ctrl-C, interrupt and throw exception to cancel prompting.
-        Thread.currentThread().interrupt();
-        throw new RuntimeException(ex);
-    }
-
-    @Override
     public int getTerminalWidth() {
         return getWidth();
     }
 
-    @Override
     public int getTerminalHeight() {
         return getHeight();
     }
 
     private int getHeight() {
+        if (connection == null) {
+            return 40;
+        }
         if (connection instanceof TerminalConnection) {
```

Comment:
why a separate checkf for TerminalConnection? cant we just call connection.size()?

Reply:
Good point, removed useless cast, I guess a leftover from earlier API.